### PR TITLE
Add high compression feature for PXC operator

### PR DESCRIPTION
**Pull request by hors**

### CODE REVIEW

File: build/pxc-configure-pxc.sh [hunk 1]
```diff
@@ -69,6 +69,10 @@ MYSQL_VERSION=$(mysqld -V | awk '{print $3}' | awk -F'.' '{print $1"."$2}')
 if [ "$MYSQL_VERSION" == '8.0' ]; then
 	grep -E -q "^[#]?admin-address" "$CFG" || sed '/^\[mysqld\]/a admin-address=\n' ${CFG} 1<>${CFG}
 	grep -E -q "^[#]?log_error_suppression_list" "$CFG" || sed '/^\[mysqld\]/a log_error_suppression_list="MY-010055"\n' ${CFG} 1<>${CFG}
+    if [ "$HIGH_COMPRESSION" == 'yes' ]; then
+        sed -r "s|^[#]?default_storage_engine=.*$|default_storage_engine=BLACKHOLE|" ${CFG} 1<>${CFG}
```
Comment (spron-in) [inline build/pxc-configure-pxc.sh:73]: This is awesome! Very neat solution.
I assume it reduces the time to backup and restore to almost 0.

Comment (dAdAbird) [inline build/pxc-configure-pxc.sh:73]: We have the same in PBM https://github.com/percona/percona-backup-mongodb/tree/main/pbm/storage/blackhole
Exciting and effective stuff, but careful using it. We've lost a few clusters over the event horizon.

File: pkg/pxc/app/statefulset/node.go [hunk 2]
```diff
@@ -156,6 +156,14 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaXt
 			},
 		})
 	}
+
+	if cr.CompareVersionWith("1.8.0") >= 0 && *cr.Spec.PXC.HighCompression {
+		appc.Env = append(appc.Env, corev1.EnvVar{
+			Name:  "HIGH_COMPRESSION",
```
Comment (dAdAbird) [inline pkg/pxc/app/statefulset/node.go:162]: `HIGH_COMPRESSION` doesn't fully reflect properties.
I suugest `VERY_VERY_HIGH_COMPRESSION`

File: build/pxc-configure-pxc.sh (hunk 1 above)
Comment (dAdAbird) [reply build/pxc-configure-pxc.sh:73]: And I don't fully understand which type of black hole is used?
Schwarzschild black hole would grow in mass (hence maintenance and retention issues).
And with Kerr black holes there is a possibility of some time-related issues (see. https://arxiv.org/pdf/0706.0622v3.pdf) 
I suggest proper testing.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | failed |
| haproxy | failed |
| big-data | failed |
| upgrade-haproxy | failed |
| storage | failed |
| recreate | failed |
| self-healing | failed |
| scaling-proxysql | failed |
| init-deploy | failed |
| restore-to-encrypted-cluster | failed |
| self-healing-advanced | failed |
| security-context | failed |
| demand-backup | failed |
| upgrade-proxysql | failed |
| limits | failed |
| operator-self-healing | failed |
| pitr | failed |
| monitoring-2-0 | failed |
| demand-backup-encrypted-with-tls | failed |
| affinity | failed |
| one-pod | failed |
| smart-update | failed |
| scheduled-backup | failed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |
| upgrade-consistency | failed |
| users | failed |
| tls-issue-self | failed |
| tls-issue-cert-manager | failed |
| tls-issue-cert-manager-ref | failed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/828/commits/5a932efc7492f9721849085e3ecd9be29b3c0216
image: `perconalab/percona-xtradb-cluster-operator:PR-828-5a932efc`

**Comment by delgod** [comment]
it was the 1st of April joke

**Pull request closed without merging**